### PR TITLE
new NOT WORKING machines (ABL Pinball plug & play)

### PR DESCRIPTION
**Pull request by DavidHaywood**
new NOT WORKING machines
-----
Pinball (P8002, ABL TV Game) [David Haywood,  Morten Kirkegaard, Peter Wilhelmsen]

### CODE REVIEW

File: src/mame/drivers/nes_vt.cpp [hunk 1]
```diff
@@ -1417,6 +1528,32 @@ void nes_vt_state::nes_vt_base(machine_config &config)
...
+
+		PPU_VT03PAL(config.replace(), m_ppu);
+		m_ppu->set_cpu_tag(m_maincpu);
+		m_ppu->int_callback().set_inputline(m_maincpu, INPUT_LINE_NMI);
+		m_ppu->read_bg().set(FUNC(nes_vt_state::chr_r));
+		m_ppu->read_sp().set(FUNC(nes_vt_state::spr_r));
+
+		/* video hardware */
+		m_screen->set_refresh_hz(50.0070);
+		m_screen->set_vblank_time(ATTOSECONDS_IN_USEC((106.53 / (PAL_APU_CLOCK.dvalue() / 1000000)) *
```
Comment (kazblox) [inline src/mame/drivers/nes_vt.cpp]: Should be m_screen->set_vblank_time(ATTOSECONDS_IN_USEC((113.66/(PALC_APU_CLOCK.dvalue()/1000000)) * (ppu2c0x_device::VBLANK_LAST_SCANLINE_PAL-ppu2c0x_device::VBLANK_FIRST_SCANLINE_PALC+1+2)));

File: src/mame/drivers/nes_vt.cpp [hunk 2]
```diff
@@ -1417,6 +1528,32 @@ void nes_vt_state::nes_vt_base(machine_config &config)
 	m_apu->add_route(ALL_OUTPUTS, "mono", 0.50);
 }
 
+void nes_vt_ablpinb_state::nes_vt_ablpinb(machine_config &config)
+{
+	nes_vt_base(config);
+
+	if (0)	// the dumped ABL Pinball unit is PAL (26.6017 XTAL) although Asian ones are likely NTSC.  PAL NES emulation seems more broken at the moment, probably timing issues in the PPU
+	{
+		m_maincpu->set_clock(PAL_APU_CLOCK);
```
Comment (kazblox) [inline src/mame/drivers/nes_vt.cpp]: Should be PALC_APU_CLOCK.

File: src/devices/video/ppu2c0x_vt.cpp [hunk 3]
```diff
@@ -16,18 +16,31 @@
...
+ppu_vt03_device::ppu_vt03_device(const machine_config& mconfig, const char* tag, device_t* owner, uint32_t clock)
+	: ppu_vt03_device(mconfig, PPU_VT03, tag, owner, clock)
+{
+}
+
+
+ppu_vt03pal_device::ppu_vt03pal_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	ppu_vt03_device(mconfig, PPU_VT03PAL, tag, owner, clock)
+{
+	m_scanlines_per_frame = PAL_SCANLINES_PER_FRAME;
```
Comment (kazblox) [inline src/devices/video/ppu2c0x_vt.cpp]: Add "m_vblank_first_scanline = VBLANK_FIRST_SCANLINE_PALC;".

**Comment by kazblox** [comment]
Explanation for the comments: PAL VT PPU should be modeled after ppu_palc, not ppu_pal.

File: src/devices/video/ppu2c0x_vt.h [hunk 4]
```diff
@@ -26,6 +26,7 @@ enum vtxx_pal_mode {
 class ppu_vt03_device : public ppu2c0x_device {
 public:
 	ppu_vt03_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock = 0);
+	ppu_vt03_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock = 0);
```
Comment (cuavas) [inline src/devices/video/ppu2c0x_vt.h]: Trampoline constructors should be protected, and the clock should never be defaulted in a trampoline constructor.

Also, all these PPUs should have their clocks documented - they definitely aren't clockless.  Defaulting the clocks to zero in the public constructors makes life easier but it hurts documentation,

File: src/mame/drivers/nes_vt.cpp [hunk 5]
```diff
@@ -1417,6 +1528,32 @@ void nes_vt_state::nes_vt_base(machine_config &config)
 	m_apu->add_route(ALL_OUTPUTS, "mono", 0.50);
 }
 
+void nes_vt_ablpinb_state::nes_vt_ablpinb(machine_config &config)
+{
+	nes_vt_base(config);
+
+	if (0)	// the dumped ABL Pinball unit is PAL (26.6017 XTAL) although Asian ones are likely NTSC.  PAL NES emulation seems more broken at the moment, probably timing issues in the PPU
```
Comment (cuavas) [inline src/mame/drivers/nes_vt.cpp]: Why would Asian ones be NTSC?  Most of Asia uses PAL (besides Japan with NTSC and some SECAM holdouts).

Comment (DavidHaywood) [reply src/mame/drivers/nes_vt.cpp]: ah ok, I thought it was an NTSC region as I'm used to seeing NTSC-C for China, although that appears to be a 2003+ thing.  some of the Asian units we've seen have had NTSC clocks tho.

Comment (cuavas) [reply src/mame/drivers/nes_vt.cpp]: Sony regions are weird.  NTSC-C what Sony calls the copy protection region for PRC excluding Hong Kong and Macau.  The consoles output PAL video.  Sony includes Hong Kong and Macau in the NTSC-J region for copy protection purposes, although TVs in these territories also use PAL standard.

Comment (DavidHaywood) [reply src/mame/drivers/nes_vt.cpp]: anyway, i'll just set it to the PAL clone timings as mixmaster put forward as unlike plain PAL this seems to leave it in a stable enough state to try and figure out what they've done with the inputs, which is very much non-standard here.

File: src/devices/video/ppu2c0x_vt.h (hunk 4 above)
Comment (DavidHaywood) [reply src/devices/video/ppu2c0x_vt.h]: well I'll set them in the driver to use the PAL/NTSC clocks then, as there are no other XTALs on these.  Not 100% convinved that potentially bad documentation is better than just having at 0 (and this pattern is widespread in the existing NES code) but I'll try to keep the part I'm changing in check at least.

**Pull request merged**